core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -653,6 +684,8 @@ public int hashCode() {
         result = 31 * result + (currentNodeId != null ? currentNodeId.hashCode() : 0);
         result = 31 * result + (relocatingNodeId != null ? relocatingNodeId.hashCode() : 0);
         result = 31 * result + (primary ? 1 : 0);
+        result = 31 * result + (int) (primaryTerm ^ (primaryTerm >>> 32));
```

Comment:
Another place for [`Long.hashCode`](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#hashCode-long-)?